src/main/java/com/google/codeu/servlets/BobaDataServlet.java

```diff
@@ -0,0 +1,80 @@
+package com.google.codeu.servlets;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import java.io.IOException;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Returns boba shop data as a JSON array, e.g. [{"lat": 38.4404675, "lng": -122.7144313}].
+ */
+@WebServlet("/boba-data")
+public class BobaDataServlet extends HttpServlet {
+  private int nameIdx = 2;
+  private int ratingIdx = 3;
+  private int addressIdx = 4;
+  private int cityIdx = 5;
+  private int latIdx = 6;
+  private int longIdx = 7;
+
+  JsonArray bobaShopsArray;
+
+  @Override
+  public void init() {
+    this.bobaShopsArray = parseData();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    response.getOutputStream().println(bobaShopsArray.toString());
+  }
+
+  /**
+   * Read and parse data from the csv file into JsonArray.
+   */
+  private JsonArray parseData() {
```

Comment:
"parseData" is a vague name. What data? Where does it come from?

I would suggest splitting this function into two:

1. Static "parseCSV", which would take Scanner and return JsonArray.
2. getCSV or getStaticCSV, which would load the file and return scanner.